pkg/services/sqlstore/migrations/entity_events_mig.go

```diff
@@ -0,0 +1,18 @@
+package migrations
+
+import . "github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+
+func addEntityEventsTableMigration(mg *Migrator) {
+	entityEventsTable := Table{
+		Name: "entity_event",
+		Columns: []*Column{
+			{Name: "id", Type: DB_BigInt, Nullable: false, IsPrimaryKey: true, IsAutoIncrement: true},
+			{Name: "grn", Type: DB_NVarchar, Length: 1024, Nullable: false},
+			{Name: "event_type", Type: DB_NVarchar, Length: 8, Nullable: false},
+			{Name: "created", Type: DB_BigInt, Nullable: false},
+		},
+		Indices: []*Index{},
+	}
```

Comment:
I understand this solution is build with dashboards in mind, but what's your plan when introducing support for indexing other things besides dashboards? It feels like this schema might not be optimal for that. Or maybe you're planning to throw this away when supporting search for other things besides dashboards?

If planning to support more things than dashboards might be beneficial to store the entity type, e.g. `dashboard`. Maybe I would suggest storing this in the `entity_type` column. Thinking of for example [GitHub events](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads) their using event type for `issues`, `pull_request` etc and then they have an `action` field telling what the event was all about, e.g. created, updated, deleted etc. At least this fields more forward safe to me, IF again there's a plan to supporting more things besides dashboards.

Not sure I understand the idea with a column `grn`, maybe you can explain that a bit more. I've read a couple design docs referring to this thing, but not sure it's worth introducing it here?

Reply:
So till now we supposed that `entity_type` can be extracted from GRN, we can add separate column for it too - but more like a "safety bag" thing.

Reply:
> It is most likely going to include

and org ID probably. But maybe we will find a way to avoid this.

Reply:
> So till now we supposed that `entity_type` can be extracted from GRN, we can add separate column for it too - but more like a "safety bag" thing.

> and org ID probably. But maybe we will find a way to avoid this (like matching storages to orgs).

Got it :+1: Think what's the most important is to find a consistent way expressing (whatever format we decide) in the URL, that doesn't necessarily be exactly the same as the internal representation. Guess that was my main point regarding naming the column `grn` when you're not sure where the future will bring you then maybe something more general like `key` could have been used - since the service owning the database table it have the knowledge of what is stored and how. Definitely no blocker, just wanted to explain my thoughts a bit more.

Reply:
That makes sense total sense - I think `entity_id` is a good fit and also better matches the table name